### PR TITLE
Some more F+O 3.1 functions

### PR DESCRIPTION
**Pull request by adamretter**
Implements from XPath Functions and Operators 3.0:

```xquery
fn:has-children() as xs:boolean
fn:has-children($node as node()?) as xs:boolean
fn:innermost($nodes as node()*) as node()*
fn:outermost($nodes as node()*) as node()*
```

### CODE REVIEW

File: src/org/exist/xquery/functions/fn/FnModule.java [hunk 1]
```diff
@@ -122,6 +122,8 @@
         new FunctionDef(FunAdjustTimezone.fnAdjustTimeToTimezone[1], FunAdjustTimezone.class),
         new FunctionDef(FunAdjustTimezone.fnAdjustDateTimeToTimezone[0], FunAdjustTimezone.class),
         new FunctionDef(FunAdjustTimezone.fnAdjustDateTimeToTimezone[1], FunAdjustTimezone.class),
+        new FunctionDef(FnHasChildren.FNS_HAS_CHILDREN_0, FnHasChildren.class),
```
Comment (dizzzz) [inline src/org/exist/xquery/functions/fn/FnModule.java]: Reference to InnerMost function is missing?

File: test/src/xquery/fn.xql [hunk 2]
```diff
@@ -21,4 +21,52 @@ declare
 function fnt:normalize-unicode($normalization-form as xs:string) {
     count(string-to-codepoints(normalize-unicode($fnt:composed, $normalization-form))),
     count(string-to-codepoints(normalize-unicode($fnt:decomposed, $normalization-form)))
+};
```
Comment (dizzzz) [inline test/src/xquery/fn.xql]: InnerMost tests are missing?

**Comment by adamretter** [comment]
@dizzzz ah okay give me a sec...

**Comment by dizzzz** [comment]
:)

**Comment by dizzzz** [comment]
@adamretter ?

**Comment by adamretter** [comment]
@dizzzz Okay I refactored this and added the missing code. Should be ready to land now :-)

**Comment by joewiz** [comment]
Whoo hoo! 4 functions closer to full coverage of XQuery 3.1 F&O! Thanks @adamretter!

**Pull request merged**